### DISCUSSION TITLE
Ruby 2.6.0 installation fails on MacOS Mojave and Catalina both.

### DISCUSSION THREAD

**Discussion by marvindanig** [rbenv/ruby-build#1845] [Build failures]
Here's the question on [SO](https://stackoverflow.com/questions/59483699/ruby-2-6-0-installation-fails-on-macos-with-rbenv) with logs pertaining to Mojave. The ones below pertain to MacOS Catalina:

```
rbenv install 2.6.5
Downloading openssl-1.1.1d.tar.gz...
-> https://dqw8nmjcqpjn7.cloudfront.net/1e3a91bc1f9dfce01af26026f856e064eab4c8ee0a8f457b5ae30b40b8b711f2
Installing openssl-1.1.1d...

BUILD FAILED (OS X 10.15.2 using ruby-build 20191225)

Inspect or clean up the working tree at /var/folders/l_/xgbzwc3j0xn_cb74550m71vr0000gn/T/ruby-build.20191226144551.46445.ELb9GY
Results logged to /var/folders/l_/xgbzwc3j0xn_cb74550m71vr0000gn/T/ruby-build.20191226144551.46445.log

Last 10 log lines:
      _s_server_main in s_server.o
  "_verify_stateless_cookie_callback", referenced from:
      _s_server_main in s_server.o
  "_wait_for_async", referenced from:
      _s_client_main in s_client.o
      _sv_body in s_server.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [apps/openssl] Error 1
make: *** [all] Error 2

```

Read somewhere that someone has pushed in C++ code on a plain C program somewhere. Don't know how much of that is true.

**Comment by marvindanig** [comment]
Here's the full error log as [gist](https://gist.github.com/marvindanig/97e3d7175403e6b630a9a79c1caf8dbc).

**Comment by marvindanig** [comment]
I noticed the same issue on another app of mine with ruby 2.5.1 and rails 5.2.0 as well. Has something to do with 32/64 bit machines and the gems that seem to pick the bug up are:

1. Typhoeus
2. Bootsnap
3. Saasc
4. Ethon.

I posted the issue on [stack overflow](https://stackoverflow.com/questions/59512555/rails-t-rails-server-puma-is-failing).

**Comment by whoisjake** [comment]
Getting this trying to install 2.7.0 on macOS 10.14.6 (Mojave)... 
Same build error, pretty exact build log as well.

**Comment by wadkar** [comment]
It looks like the issue is with building openssl, try installing openssl [as suggested in the wiki](https://github.com/rbenv/ruby-build/wiki#suggested-build-environment):

```sh
$ brew install openssl libyaml libffi
$ export RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)"
$ rbenv install 2.6.5
```

Currently, ruby-build [doesn't look for homebrew openssl](https://github.com/rbenv/ruby-build/commit/9d96a6b83db489941c5a6499d7f765e3d13bedc2) so the above solution might work by explicitly telling it to do so.
